src/Deku/Interpret.js

```diff
@@ -326,23 +500,58 @@ export const makeRoot_ = (a) => (state) => () => {
 	};
 };
 
-export const giveNewParent_ = (a) => (state) => () => {
-	if (state.units[a.id]) {
-
-		var ptr = a.id;
-		var parent = a.parent;
+export const giveNewParent_ = (runOnJust) => (b) => (state) => () => {
+	const runMe = []
+	if (state.units[b.id] && state.units[b.id].startBeacon) {
+		var c = b;
+		while (c) {
+			runMe.push(c);
+			c = c.nextSibling;
+			if (c === state.units[b.id].endBeacon) {
+				break;
+			}
+		}
+	} else {
+		runMe.push(b);
+	}
+	for (var z = 0; z < runMe.length; z++) {
+		const a = runMe[z]
```

Comment:
missing semicolon – what do you think this is, coffeescript?? 😉

Reply:
☕